### PR TITLE
Npc and freeze fix

### LINKED ISSUE #168: NPC entities not displaying
(Tested on 3 clients)
Currently attempting to create your own NPC entity will result with no reaction. Copying a existing NPC entity will work fine, until the NPC is edited, from then on the editor will act as though it is not there when re loading the map.

### PR DESCRIPTION
**Pull request by 2767mr**
Fixes #168 
* Changed `Helper.getJson` to return undefined if no file was found.
* Fixed infinite loop in SideNav.

### CODE REVIEW

File: webapp/src/app/components/sidenav/sidenav.component.ts [hunk 1]
```diff
@@ -35,22 +35,11 @@ export class SidenavComponent implements OnInit {
...
-			this.currentView = view;
-			switch (view) {
-				case EditorView.Layers:
-					this.activeTab = 0;
-					break;
-				case EditorView.Entities:
-					this.activeTab = 1;
-					break;
+			if (view !== undefined && this.activeTab != view) {
+				this.activeTab = view;
```
Comment (Vegita2) [inline webapp/src/app/components/sidenav/sidenav.component.ts:42]: Missed this the first time. `activeTab` is a number and view is not, should be changed to match.

Comment (2767mr) [reply webapp/src/app/components/sidenav/sidenav.component.ts:42]: Fixed

File: webapp/src/app/components/sidenav/sidenav.component.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ import {MatTabChangeEvent} from '@angular/material';
 })
 export class SidenavComponent implements OnInit {
 	
-	activeTab = 0;
+	activeTab = EditorView.Layers;
```
Comment (Vegita2) [inline webapp/src/app/components/sidenav/sidenav.component.ts:17]: If `activeTab` is changed to EditorView then it is the same as `currentView`. In this case the `currentView` variable could be removed

File: webapp/src/app/components/sidenav/sidenav.component.html [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <div fxLayout="column" class="nav-container" *ngIf="tilemap">
-	<mat-tab-group (selectedTabChange)="tabChanged($event)" [selectedIndex]="activeTab">
+	<mat-tab-group (selectedTabChange)="tabChanged($event)" [(selectedIndex)]="activeTab">
```
Comment (Vegita2) [inline webapp/src/app/components/sidenav/sidenav.component.html:2]: Is the `selectedIndex` event really necessary? In this case activeTab is set twice, once from `(selectedTabChange)` and once from `(selectedIndex)`

File: webapp/src/app/components/sidenav/sidenav.component.ts [hunk 4]
```diff
@@ -35,22 +35,11 @@ export class SidenavComponent implements OnInit {
...
 		this.globalEvents.currentView.subscribe(view => {
-			this.currentView = view;
-			switch (view) {
-				case EditorView.Layers:
-					this.activeTab = 0;
-					break;
-				case EditorView.Entities:
-					this.activeTab = 1;
-					break;
+			if (view !== undefined && this.activeTab != view) {
```
Comment (Vegita2) [inline webapp/src/app/components/sidenav/sidenav.component.ts:41]: lint: ` != should be !==`

File: webapp/src/app/components/sidenav/sidenav.component.ts (hunk 2 above)
Comment (2767mr) [reply webapp/src/app/components/sidenav/sidenav.component.ts:17]: `currentView` is in fact unused. Fixed by removal in fdd25bafcbf6f475002ed4916aa8757f3d4bd3cc.

File: webapp/src/app/components/sidenav/sidenav.component.ts (hunk 4 above)
Comment (2767mr) [reply webapp/src/app/components/sidenav/sidenav.component.ts]: Fixed in 091d376bb3c9cdc9d1c93f4672df0b4a50ef2107

File: webapp/src/app/components/sidenav/sidenav.component.html (hunk 3 above)
Comment (2767mr) [reply webapp/src/app/components/sidenav/sidenav.component.html]: The binding seems to be unnecessary. Removed in 6ead0628ebbbc7ece8c82c7d0f47a99749600658.

**Pull request merged**